### DISCUSSION TITLE
Memory leak when using Managed Client with TLS in v4.1.2 and v4.1.3

### DISCUSSION THREAD

**Discussion by brslt** [dotnet/MQTTnet#1605] [Q&A] [answered]
Hi, we use the Managed Client for subscribing to data from two different sources. For one of them, we use TLS, for the other one, basic auth. When using v4.1.0, this worked fine. However, when we upgraded to v4.1.2 (also in v4.1.3) we experienced a memory leak that after some hours caused our application to crash. We pinpointed the memory leak to be caused by the client using TLS, but we didn't dive deeper into it. When reverting to v4.1.0, everything worked fine again.

**Comment by chkr1011** [comment, marked as answer]
I doubt that this is related to TLS because we simply handover all the TLS stuff to the .NET Framework. But we found another memory leak in the versions you mentioned. Please try (exactly) version 4.1.4.524 from the myget feed. It includes a fix for the memory leak. 

Please let me know if it works. If not, please share the type of objects which is leaking.

**Reply by brslt** [reply]
OK, thanks! We will check it out.

**Comment by brslt** [comment]
Hi again, we have now for 24 hours been running v4.1.4.524 in our staging environment (with close to production payload), and the memory consumption has been smooth and stable just like we experienced with v4.1.0. So thumbs up from our side!
